File: python-example/pages/registration_page.py
from selenium.webdriver.support.wait import WebDriverWait
from selenium.webdriver.support.select import Select


class RegistrationPage:

    def __init__(self, driver):
        self.driver = driver
        self.wait = WebDriverWait(driver, 10)

    def open(self):
        self.driver.get("http://localhost/litecart/en/create_account")
        return self

    @property
    def firstname_input(self):
        return self.driver.find_element_by_name("firstname")

    @property
    def lastname_input(self):
        return self.driver.find_element_by_name("lastname")

    @property
    def address1_input(self):
        return self.driver.find_element_by_name("address1")

    @property
    def postcode_input(self):
        return self.driver.find_element_by_name("postcode")

    @property
    def city_input(self):
        return self.driver.find_element_by_name("city")

    @property
    def email_input(self):
        return self.driver.find_element_by_name("email")

    @property
    def phone_input(self):
        return self.driver.find_element_by_name("phone")

    @property
    def password_input(self):
        return self.driver.find_element_by_name("password")

    @property
    def confirmed_password_input(self):
        return self.driver.find_element_by_name("confirmed_password")

    @property
    def create_account_button(self):
        return self.driver.find_element_by_name("create_account")

    def select_country(self, country):
        self.driver.find_element_by_css_selector("[id ^= select2-country_code]").click()
        self.driver.find_element_by_css_selector(".select2-results__option[id $= %s]" % country).click()

    def select_zone(self, zone):
        self.wait.until(lambda d: d.find_element_by_css_selector("select[name=zone_code] option[value=%s]" % zone))
        Select(self.driver.find_element_by_css_selector("select[name=zone_code]")).select_by_value(zone)

File: java-example/src/test/java/ru/stqa/training/selenium/pages/RegistrationPage.java
package ru.stqa.training.selenium.pages;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;
import org.openqa.selenium.support.PageFactory;
import org.openqa.selenium.support.ui.Select;

public class RegistrationPage extends Page {

    public RegistrationPage(WebDriver driver) {
        super(driver);
        PageFactory.initElements(driver, this);
    }

    public void open() {
        driver.get("http://localhost/litecart/en/create_account");
    }

    @FindBy(name="firstname")
    public WebElement firstnameInput;

    @FindBy(name="lastname")
    public WebElement lastnameInput;

    @FindBy(name="address1")
    public WebElement address1Input;

    @FindBy(name="postcode")
    public WebElement postcodeInput;

    @FindBy(name="city")
    public WebElement cityInput;

    @FindBy(name="email")
    public WebElement emailInput;

    @FindBy(name="phone")
    public WebElement phoneInput;

    @FindBy(name="password")
    public WebElement passwordInput;

    @FindBy(name="confirmed_password")
    public WebElement confirmedPasswordInput;

    @FindBy(name="create_account")
    public WebElement createAccountButton;

    public void selectCountry(String country) {
        driver.findElement(By.cssSelector("[id ^= select2-country_code]")).click();
        driver.findElement(By.cssSelector(
                String.format(".select2-results__option[id $= %s]", country))).click();
    }

    public void selectZone(String zone) {
        wait.until((WebDriver d) -> d.findElement(
                By.cssSelector(String.format("select[name=zone_code] option[value=%s]", zone))));
        new Select(driver.findElement(By.name("select[name=zone_code]"))).selectByValue(zone);
    }
}
